test/map-file-tests.ts

```diff
@@ -49,26 +49,26 @@ describe('Code Segments', function () {
         reader.segments.length.should.equal(1);
 
         let info = reader.getSegmentInfoByStartingAddress('0001', 0x2838);
-        info.unitName.should.equal('output.pas');
+        info?.unitName?.should.equal('output.pas');
```

Comment:
I'm not sure these tests are the same any more: doesn't the `?.` short citruit and return `undefined` if any of the values were `undefined` here? So, before we would have failed a test if `info` was `null` or `undefined` (as it would have raised an exception). But now we'll silently ignore that failure.

In other parts of the code we've either asserted or used a helper like `unwrap()`. Paging @jeremy-rifkin as our style expert though (and to check my assumptions here!)

Reply:
Yes, that would short circuit and no error would be thrown. It was an easy way out of the TS lint error. `info` and other possibly undefined variables need to be checked for robustness. I'll be working on this and waiting for Jeremy's review too.

Reply:
Sorry for the delayed response! New changes look good, reading through them now

Reply:
`unwrap` has the advantage of telling us sooner where things go wrong, e.g. if info is undefined, but `expect(info?.unitName)` is probably fine too

Reply:
I can replace the expect() with unwrap().
We would get case like `unwrap(unwrap(info).displayName)`
I see that a few other tests are already using unwrap(), so we need to be consistent with that please let me know.

Reply:
Sounds good